ilastik/applets/edgeTraining/opEdgeTraining.py

```diff
@@ -400,10 +400,21 @@ def execute(self, slot, subindex, roi, result):
             classifier = self.EdgeClassifier.value
 
             # Classifier can be None if no labels have been selected
-            if classifier is None or len(classifier.known_classes) < 2:
+            if classifier is None or len(classifier.known_classes) == 0:
                 result[0] = np.zeros((len(edge_features_df),), dtype=np.float32)
                 return
 
+            known_classes = classifier.known_classes
+            assert set(known_classes).issubset([1, 2])
+            if len(known_classes) == 1:
```

Comment:
I think it would be easier to follow the control flow if we extract all logic into `_execute` helper method that checks various conditions and early-returns `probabilities`, instead of multiple nested conditions in the original `execute`:

```python
def _execute(self):
    if not self.TrainRandomForest.value:
        ...
        return classifier.predict_probabilities(feature_matrix)[:, 1]

    if classifier is None or ...:
        return np.zeros((len(edge_features_df),), dtype=np.float32)

    ...
    return some_default_value

def execute(self, ..., result):
    result[0] = self._execute()
```